qlight/server.go

```diff
@@ -0,0 +1,186 @@
+		if tx.IsPrivacyMarker() {
+			ptd, err := p.fetchPrivateData(tx.Data(), psm)
+			if err != nil {
+				return nil, err
+			}
+			if ptd != nil {
+				pvtTxs = append(pvtTxs, *ptd)
+			}
+
+			innerTx, _, _, _ := private.FetchPrivateTransactionWithPTM(tx.Data(), p.ptm)
+			if innerTx != nil {
+				tx = innerTx
+			}
+		}
+
+		if tx.IsPrivate() {
+			ptd, err := p.fetchPrivateData(tx.Data(), psm)
+			if err != nil {
+				return nil, err
+			}
+			if ptd != nil {
+				pvtTxs = append(pvtTxs, *ptd)
+			}
+		}
+	}
+	if len(pvtTxs) == 0 {
+		return nil, nil
+	}
+
+	var privateStateRoot = common.Hash{}
+
+	stateRepo, err := p.privateStateManager.StateRepository(block.Root())
+	if err != nil {
+		log.Debug("Unable to retrieve private state repo while preparing the private block data", "block.No", block.Number(), "psi", psi, "err", err)
+	} else {
+		privateStateRoot, err = stateRepo.PrivateStateRoot(PSI)
+		if err != nil {
+			log.Debug("Unable to retrieve private state root while preparing the private block data", "block.No", block.Number(), "psi", psi, "err", err)
+		}
+	}
+
+	return &BlockPrivateData{
+		BlockHash:           block.Hash(),
+		PSI:                 PSI,
+		PrivateStateRoot:    privateStateRoot,
+		PrivateTransactions: pvtTxs,
+	}, nil
+}
+
+func (p *privateBlockDataResolverImpl) fetchPrivateData(encryptedPayloadHash []byte, psm *mps.PrivateStateMetadata) (*PrivateTransactionData, error) {
+	txHash := common.BytesToEncryptedPayloadHash(encryptedPayloadHash)
+	_, _, privateTx, extra, err := p.ptm.Receive(txHash)
+	if err != nil {
+		return nil, err
+	}
+	// we're not party to this transaction
+	if privateTx == nil {
+		return nil, nil
+	}
+	if p.privateStateManager.NotIncludeAny(psm, extra.ManagedParties...) {
+		return nil, nil
+	}
+
+	extra.ManagedParties = psm.FilterAddresses(extra.ManagedParties...)
+
+	ptd := PrivateTransactionData{
+		Hash:     &txHash,
+		Payload:  privateTx,
+		Extra:    extra,
+		IsSender: false,
+	}
+	if len(psm.Addresses) == 0 {
+		// this is not an MPS node so we have to ask tessera
+		ptd.IsSender, _ = p.ptm.IsSender(txHash)
+	} else {
+		// this is an MPS node so we can speed up the IsSender logic by checking the addresses in the private state metadata
+		ptd.IsSender = !psm.NotIncludeAny(extra.Sender)
+	}
+
+	return &ptd, nil
+}
+
+type authProviderImpl struct {
+	privateStateManager mps.PrivateStateManager
+	authManagerProvider AuthManagerProvider
+	authManager         security.AuthenticationManager
+	enabled             bool
+}
+
+func NewAuthProvider(privateStateManager mps.PrivateStateManager, authManagerProvider AuthManagerProvider) AuthProvider {
+	return &authProviderImpl{
+		privateStateManager: privateStateManager,
+		authManagerProvider: authManagerProvider,
+		enabled:             false,
+	}
+}
+
+func (a *authProviderImpl) Initialize() error {
+	if a.authManagerProvider != nil {
+		a.authManager = a.authManagerProvider()
+		if a.authManager == nil {
+			return nil
+		}
```

Comment:
I think `provider` fits better as opposed to New/Get. New is mostly used as a constructor in Go.
Here you check if a an auth manager `provider` is available (and invoke the method if it is).

Do you have another suggestion instead of new/get?

There are quite a few other similarly named functions in other areas of the quorum code (PSI/credentials providers).